### PR TITLE
Modified input tool to reflect new schema change

### PR DESCRIPTION
**Pull request by brandonzhu09**
## Description
Input tool has been modified to parse the new filename format with the updated file structure. Enforced the reference to the readers metadata table and added a helper method to parse cycle times into a datetime object. Fixed a return message bug that incorrectly displays the connection as open when an unchecked exception has occurred.

### CODE REVIEW

**Comment by asewnath** [comment]
Hi @brandonzhu09, I'll wait to review this PR until I can test both evagram and evagram-input. Let me know when both branches are ready to be tested

**Comment by brandonzhu09** [comment]
> Hi @brandonzhu09, I'll wait to review this PR until I can test both evagram and evagram-input. Let me know when both branches are ready to be tested

Hi Akira, so far, my progress this week has been to finalize development and testing of evagram-input to insert new diagnostics under the updated schema. I updated the evagram-input tests to check for edge cases like an incorrect reader name, plot type, and cycle time in the wrong format. Now I am turning my focus back to evagram but now with updating the frontend with the new dropdown options and also adding in the API endpoints to interface with the dropdowns. 

Do you know if it is a good idea to add those changes in the `feature/readers_schema_update` branch in evagram? Because if I did, it won't be just the schema that I am updating in the feature. I just don't want the branch to get so big like last time when it can be its own separate branch.

**Comment by asewnath** [comment]
@brandonzhu09 Got it, okay you can make those dropdown changes in a separate branch.

File: src/evagram_input/input_tool/input_tool.py [hunk 1]
```diff
@@ -131,22 +144,26 @@ def _add_current_experiment(self, experiment_name, owner_id):
...
-        if len(current_experiment) == 1:
+        current_experiment = self._cursor.fetchone()
+        if current_experiment:
             # returns experiment_id of current experiment
-            return current_experiment[0][0]
+            return current_experiment[0]
         else:
+            # generate creation date for experiment
+            timezone = pytz.timezone("US/Eastern")
+            create_date = datetime.now(timezone)
```
Comment (asewnath) [inline src/evagram_input/input_tool/input_tool.py:154]: Although this isn't a problem for now, let's avoid assuming that timezone will always be `US/Eastern`. Consider using `DateTime.UtcNow`

Comment (brandonzhu09) [reply src/evagram_input/input_tool/input_tool.py:154]: Thanks for catching this, I was going to ask about this but now the idea is to convert all the timestamps to UTC time? It also appears that `datetime.utcnow()` is deprecated since version 3.12 so I will use `datetime.now(timezone.utc)`.

Comment (asewnath) [reply src/evagram_input/input_tool/input_tool.py:154]: Yeah for now convert timestamps to UTC

**Review by asewnath — APPROVED** [review]
